### PR TITLE
Log non-JSON slicec output as a regular error

### LINKED ISSUE #4475: [Audit-Medium] malformed or non-JSON slicec output can crash the MSBu…
> **AI-generated audit finding** — this issue was opened from an automated security/correctness audit. It **has not been triaged by a human yet**; verify the reasoning, reproducibility, and severity before acting on it.

### Medium: malformed or non-JSON `slicec` output can crash the MSBuild task instead of surfacing a normal compiler error — **CONFIRMED, NARROWED**

Affected code:

- [src/IceRpc.Slice.Tools/SlicecTask.cs:80](https://github.com/icerpc/icerpc-csharp/blob/150d6ee01de9609eb037963cf2897df190c23809/src/IceRpc.Slice.Tools/SlicecTask.cs#L80) — forces `--diagnostic-format=json`
- [src/IceRpc.Slice.Tools/SlicecTask.cs:106](https://github.com/icerpc/icerpc-csharp/blob/150d6ee01de9609eb037963cf2897df190c23809/src/IceRpc.Slice.Tools/SlicecTask.cs#L106) — deserializes every output line as `Diagnostic` with no `try/catch`
- [src/IceRpc.Slice.Tools/SlicecTask.cs:117](https://github.com/icerpc/icerpc-csharp/blob/150d6ee01de9609eb037963cf2897df190c23809/src/IceRpc.Slice.Tools/SlicecTask.cs#L117) — iterates `diagnostic.Notes` directly
- [src/IceRpc.Slice.Tools/Diagnostic.cs:9-17](https://github.com/icerpc/icerpc-csharp/blob/150d6ee01de9609eb037963cf2897df190c23809/src/IceRpc.Slice.Tools/Diagnostic.cs#L9) — local DTO defaults missing fields to empty strings / empty arrays rather than enforcing required fields

Verification:

Confirmed, but narrower than the original report stated.

The hard failure mode is real: if `slicec` emits a line that is not valid JSON, `JsonSerializer.Deserialize<Diagnostic>` throws and the exception escapes `LogEventsFromTextOutput`. This turns a compiler/process failure into an MSBuild task crash instead of a normal diagnostic. The same applies to semantically malformed JSON that deserializes into `diagnostic.Notes == null`, because the subsequent `foreach` will throw.

What I did **not** confirm is the broader "any schema drift crashes the task" claim. The DTO used here is package-local and intentionally permissive: `Message`, `Severity`, `ErrorCode`, and `Notes` all have defaults. A valid but unexpected JSON envelope is therefore more likely to degrade into a poor log entry, such as an empty warning, than to throw immediately.

Impact:

- `slicec` panics or malformed stderr/stdout lines surface as opaque task failures.
- Some malformed-but-valid JSON payloads can still produce misleading empty diagnostics instead of the original compiler output.

Recommendation:

- Wrap the deserialize path in `try/catch (JsonException)` and fall back to `Log.LogError(singleLine)`.
- Treat `diagnostic.Notes == null` or empty/unknown severities as malformed diagnostics and log the raw line instead of continuing.
- Add targeted tests for a non-JSON line, a valid JSON line with `notes: null`, and a valid JSON line with missing severity/message fields.

Status: **Valid, Medium severity (narrowed from the original wording)**.

---

**Source report:** [src-IceRpc.Slice.Tools-audit-2026-04-14.md](https://github.com/icerpc/icerpc-csharp-audit/blob/main/src-IceRpc.Slice.Tools-audit-2026-04-14.md) (finding `malformed or non-JSON `slicec` output can crash the MSBuild task instead of surfacing a normal compiler error — **CONFIRMED, NARROWED**`)

**Severity (auditor-assigned):** Medium

### PR DESCRIPTION
**Pull request by pepone**
Fixes #4475.

### CODE REVIEW

File: src/IceRpc.Slice.Tools/SlicecTask.cs [hunk 1]
```diff
@@ -103,8 +103,11 @@ protected override string GenerateFullPathToTool()
     /// </summary>
     protected override void LogEventsFromTextOutput(string singleLine, MessageImportance messageImportance)
     {
-        if (JsonSerializer.Deserialize<Diagnostic>(singleLine, _jsonSerializeOptions) is Diagnostic diagnostic)
+        try
         {
+            Diagnostic? diagnostic = JsonSerializer.Deserialize<Diagnostic>(singleLine, _jsonSerializeOptions);
+            Debug.Assert(diagnostic is not null);
```
[the remark this answers is not in the document]
Comment (pepone) [reply src/IceRpc.Slice.Tools/SlicecTask.cs]: fixed

**Review by externl — APPROVED** [review]

**Review by InsertCreativityHere — APPROVED** [review]
Looks good to me!

Related to #4504. Which was really a bug in `slicec`, but the tooling could of handled the `slicec` bug better; like this PR does!
